src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java

```diff
@@ -122,26 +121,32 @@ public static IToken assignRequestDefault(final IStandardRequestManager manager,
 
         request.setState(new WrappedStaticStateRequestManager(manager), RequestState.ASSIGNING);
 
-        Set<TypeToken> requestTypes = ReflectionUtils.getSuperClasses(request.getRequestType());
+        @SuppressWarnings(RAWTYPES) final Set<TypeToken> requestTypes = ReflectionUtils.getSuperClasses(request.getRequestType());
         requestTypes.remove(TypeConstants.OBJECT);
 
-        Set<IToken> failedResolvers = new HashSet<>();
+        final Set<IToken<?>> failedResolvers = new HashSet<>();
 
-        for (TypeToken requestType : requestTypes)
+        for (@SuppressWarnings(RAWTYPES) final TypeToken requestType : requestTypes)
         {
-            if (!manager.getRequestClassResolverMap().containsKey(requestType))
+            if (!manager.getRequestableTypeRequestResolverAssignmentDataStore().getAssignments().containsKey(requestType))
             {
                 continue;
             }
 
-            Collection<IRequestResolver> resolversForRequestType = manager.getRequestClassResolverMap().get(requestType);
+            Collection<IRequestResolver<?>> resolversForRequestType = manager.getRequestableTypeRequestResolverAssignmentDataStore()
+                                                                        .getAssignments()
+                                                                        .get(requestType)
+                                                                        .stream()
+                                                                        .map(iToken -> ResolverHandler.getResolver(manager, iToken))
+                                                                        .collect(Collectors.toList());
+
             resolversForRequestType = resolversForRequestType.stream()
-                                        .filter(r -> r.getRequestType().getRawType().isAssignableFrom(request.getRequestType().getRawType()))
+                                        .filter(r -> r.getRequestType().isSupertypeOf(request.getRequestType()))
                                         .filter(r -> !failedResolvers.contains(r.getRequesterId()))
                                         .sorted(Comparator.comparing(r -> -1 * r.getPriority()))
                                         .collect(Collectors.toSet());
 
-            for (final IRequestResolver resolver : resolversForRequestType)
+            for (final IRequestResolver<?> resolver : resolversForRequestType)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)